src/core/qgsnewsfeedparser.cpp

```diff
@@ -0,0 +1,298 @@
+  settings.setValue( QStringLiteral( "%1/imageUrl" ).arg( baseSettingsKey ), entry.imageUrl, QgsSettings::Core );
+  settings.setValue( QStringLiteral( "%1/content" ).arg( baseSettingsKey ), entry.content, QgsSettings::Core );
+  settings.setValue( QStringLiteral( "%1/link" ).arg( baseSettingsKey ), entry.link, QgsSettings::Core );
+  settings.setValue( QStringLiteral( "%1/sticky" ).arg( baseSettingsKey ), entry.sticky, QgsSettings::Core );
+}
+
+void QgsNewsFeedParser::fetchImageForEntry( const QgsNewsFeedParser::Entry &entry )
+{
+  // start fetching image
+  QgsNetworkContentFetcher *fetcher = new QgsNetworkContentFetcher();
+  connect( fetcher, &QgsNetworkContentFetcher::finished, this, [ = ]
+  {
+    auto findIter = std::find_if( mEntries.begin(), mEntries.end(), [entry]( const QgsNewsFeedParser::Entry & candidate )
+    {
+      return candidate.key == entry.key;
+    } );
+    if ( findIter != mEntries.end() )
+    {
+      const int entryIndex = static_cast< int >( std::distance( mEntries.begin(), findIter ) );
+
+      QImage img = QImage::fromData( fetcher->reply()->readAll() );
+
+      QSize size = img.size();
+      bool resize = false;
+      if ( size.width() > 250 )
+      {
+        size.setHeight( static_cast< int >( size.height() * static_cast< double >( 250 ) / size.width() ) );
+        size.setWidth( 250 );
+        resize = true;
+      }
+      if ( size.height() > 177 )
+      {
+        size.setWidth( static_cast< int >( size.width() * static_cast< double >( 177 ) / size.height() ) );
+        size.setHeight( 177 );
+        resize = true;
+      }
+      if ( resize )
+        img = img.scaled( size, Qt::IgnoreAspectRatio, Qt::SmoothTransformation );
+
+      //nicely round corners so users don't get paper cuts
+      QImage previewImage( size, QImage::Format_ARGB32 );
+      previewImage.fill( Qt::transparent );
+      QPainter previewPainter( &previewImage );
+      previewPainter.setRenderHint( QPainter::Antialiasing, true );
+      previewPainter.setRenderHint( QPainter::SmoothPixmapTransform, true );
+      previewPainter.setPen( Qt::NoPen );
+      previewPainter.setBrush( Qt::black );
+      previewPainter.drawRoundedRect( 0, 0, size.width(), size.height(), 8, 8 );
+      previewPainter.setCompositionMode( QPainter::CompositionMode_SourceIn );
+      previewPainter.drawImage( 0, 0, img );
+      previewPainter.end();
+
+      // Save image, so we don't have to fetch it next time
+      const QString previewDir = QStringLiteral( "%1/previewImages" ).arg( QgsApplication::qgisSettingsDirPath() );
+      QDir().mkdir( previewDir );
+      const QString imagePath = QStringLiteral( "%1/%2.png" ).arg( previewDir ).arg( entry.key );
+      previewImage.save( imagePath );
+
+      mEntries[ entryIndex ].image = QPixmap::fromImage( previewImage );
+      this->emit imageFetched( entry.key, mEntries[ entryIndex ].image );
```

Comment:
```suggestion
      emit imageFetched( entry.key, mEntries[ entryIndex ].image );
```

is this required for some reason or did you change to the pythonista first camp?

Reply:
It's needed inside lambdas where "this" is captured.